### PR TITLE
Fix _isNegative_ assignment

### PR DESCRIPTION
**Pull request by sffc**
Fixes #67

Replaces #118

### CODE REVIEW

File: numberformat/proposed.emu [hunk 1]
```diff
@@ -180,8 +180,9 @@
         1. If _x_ is ~negative-zero~, then
           1. Let _isNegative_ be *true*.
           1. Let _x_ be the mathematical value 0.
-        1. Assert: _x_ is a mathematical value.
-        1. If _x_ &lt; 0, let _isNegative_ be *true*; else let _isNegative_ be *false*.
+        1. Else,
+          1. Assert: _x_ is a mathematical value.
+          1. If _x_ &lt; 0, let _isNegative_ be *true*; else let _isNegative_ be *false*.
```
Comment (sffc) [inline numberformat/proposed.emu:185]: Given that `_x_` is a MV now, should this be `If _x_ &lt; 0` or `If ℝ(_x_) &lt; 0` ?

@gibson042

File: numberformat/proposed.emu [hunk 2]
```diff
@@ -698,8 +698,9 @@
         1. If _x_ is ~negative-zero~, then
           1. Let _isNegative_ be *true*.
           1. Let _x_ be the mathematical value 0.
```
Comment (gibson042) [inline numberformat/proposed.emu:700]:
```suggestion
          1. Set _x_ to 0.
```

File: numberformat/diff.emu [hunk 3]
```diff
@@ -714,8 +714,9 @@
         1. <ins class="block">If _x_ is ~negative-zero~, then</ins>
           1. <ins class="block">Let _isNegative_ be *true*.</ins>
           1. <ins class="block">Let _x_ be the mathematical value 0.</ins>
```
Comment (gibson042) [inline numberformat/diff.emu:716]:
```suggestion
          1. <ins class="block">Set _x_ to 0.</ins>
```

File: numberformat/proposed.emu [hunk 4]
```diff
@@ -698,8 +698,9 @@
...
-        1. Assert: _x_ is a mathematical value.
-        1. If ℝ(_x_) &lt; 0, let _isNegative_ be *true*; else let _isNegative_ be *false*.
+        1. Else,
+          1. Assert: _x_ is a mathematical value.
+          1. If ℝ(_x_) &lt; 0, let _isNegative_ be *true*; else let _isNegative_ be *false*.
```
Comment (gibson042) [inline numberformat/proposed.emu:703]:
```suggestion
          1. If _x_ &lt; 0, let _isNegative_ be *true*; else let _isNegative_ be *false*.
```

File: numberformat/diff.emu [hunk 5]
```diff
@@ -714,8 +714,9 @@
...
-        1. <ins class="block">Assert: _x_ is a mathematical value.</ins>
-        1. If ℝ(_x_) &lt; 0 <del>or _x_ is *-0*<sub>𝔽</sub></del>, let _isNegative_ be *true*; else let _isNegative_ be *false*.
+        1. <ins class="block">Else,</ins>
+          1. <ins class="block">Assert: _x_ is a mathematical value.</ins>
+          1. If ℝ(_x_) &lt; 0 <del>or _x_ is *-0*<sub>𝔽</sub></del>, let _isNegative_ be *true*; else let _isNegative_ be *false*.
```
Comment (gibson042) [inline numberformat/diff.emu:719]:
```suggestion
          1. If <del>ℝ(_x_) &lt; 0 or _x_ is *-0*<sub>𝔽</sub></del><ins>_x_ &lt; 0</ins>, let _isNegative_ be *true*; else let _isNegative_ be *false*.
```

**Review by gibson042 — APPROVED** [review]
The logic is basically correct, but _x_ should be updated with "Set" rather than "Let", and ℝ is a no-op on input that is already a mathematical value.

**Pull request merged**